src/main/java/com/google/devtools/build/lib/rules/platform/ToolchainRule.java

```diff
@@ -44,13 +45,16 @@ public RuleClass build(RuleClass.Builder builder, RuleDefinitionEnvironment env)
         .removeAttribute("deps")
         .removeAttribute("data")
         .exemptFromConstraintChecking("this rule *defines* a constraint")
+        .supportsPlatforms(false)
 
         /* <!-- #BLAZE_RULE(toolchain).ATTRIBUTE(toolchain_type) -->
         The type of the toolchain, which should be the label of a toolchain_type() target.
         <!-- #END_BLAZE_RULE.ATTRIBUTE --> */
         .add(
-            attr(TOOLCHAIN_TYPE_ATTR, BuildType.NODEP_LABEL)
+            attr(TOOLCHAIN_TYPE_ATTR, BuildType.LABEL)
```

Comment:
What does the NODEP_LABEL to LABEL change signify?

Reply:
Yes, exactly. There's now an actual dependency on the toolchain type, so:
a) it must exist, and
b) the new provider can now be used.